File: LiveData - Dagger/app/src/main/java/com/kurio/mvvmtesting/ui/activity/AddEditNoteActivity.java
package com.kurio.mvvmtesting.ui.activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.NumberPicker;

import com.kurio.mvvmtesting.R;

public class AddEditNoteActivity extends AppCompatActivity {
    EditText etTitle, etDescription;
    NumberPicker priorityPicker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_edit_note);
        etDescription = findViewById(R.id.etDescription);
        etTitle = findViewById(R.id.etTitle);
        priorityPicker = findViewById(R.id.numberPicker);
        priorityPicker.setMinValue(0);
        priorityPicker.setMaxValue(10);
        if (getIntent().hasExtra("title")) {
            initData();
        }
    }


    private void initData() {
        etTitle.setText(getIntent().getStringExtra("title"));
        etDescription.setText(getIntent().getStringExtra("description"));
        priorityPicker.setValue(getIntent().getIntExtra("priority", 0));
    }

    private void saveNote() {
        Intent i = new Intent();
        i.putExtra("description", etDescription.getText().toString().trim());
        i.putExtra("title", etTitle.getText().toString().trim());
        i.putExtra("priority", priorityPicker.getValue());
        if (getIntent().hasExtra("id"))
            i.putExtra("id", getIntent().getIntExtra("id", -1));
        setResult(RESULT_OK, i);
        finish();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.add_note_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.save_note) {
            saveNote();
            return true;
        }
        return super.onOptionsItemSelected(item);

    }
}

File: LiveData (Kotlin)/app/src/main/java/com/kurio/mvvmtesting/ui/activity/AddEditNoteActivity.kt
package com.kurio.mvvmtesting.ui.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import com.kurio.mvvmtesting.Constants
import com.kurio.mvvmtesting.R
import kotlinx.android.synthetic.main.activity_add_edit_note.*

class AddEditNoteActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_edit_note)
        numberPicker.minValue = 0
        numberPicker.maxValue = 10
        if (intent.hasExtra(Constants.EXTRA_TITLE)) {
            initData()
        }
    }


    private fun initData() {
        etTitle.setText(intent.getStringExtra(Constants.EXTRA_TITLE))
        etDescription.setText(intent.getStringExtra(Constants.EXTRA_DESCRIPTION))
        numberPicker.value = intent.getIntExtra(Constants.EXTRA_PRIORITY, 0)
    }

    private fun saveNote() {
        val i = Intent()
        i.putExtra(Constants.EXTRA_DESCRIPTION, etDescription.text.toString())
        i.putExtra(Constants.EXTRA_TITLE, etTitle.text.toString())
        i.putExtra(Constants.EXTRA_PRIORITY, numberPicker.value)
        if (intent.hasExtra(Constants.EXTRA_ID))
            i.putExtra(Constants.EXTRA_ID, intent.getIntExtra(Constants.EXTRA_ID, -1))
        setResult(Activity.RESULT_OK, i)
        finish()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.add_note_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.save_note) {
            saveNote()
            return true
        }
        return super.onOptionsItemSelected(item)

    }
}
